### PR TITLE
APIv4 - Exclude disabled custom fields

### PR DESCRIPTION
**Pull request by colemanw**
Overview
----------------------------------------
Ensures that disabled custom fields do not show up in API results or in Search Kit.

Before
----------------------------------------
Disabled custom fields available in SearchKit.

After
----------------------------------------
Disabled fields hidden.

### CODE REVIEW

File: Civi/Api4/Service/Spec/SpecGatherer.php [hunk 1]
```diff
@@ -122,6 +122,7 @@ private function addCustomFields($entity, RequestSpec $spec) {
 
     $query = CustomField::get(FALSE)
       ->setSelect(['custom_group_id.name', 'custom_group_id.title', '*'])
+      ->addWhere('is_active', '=', TRUE)
       ->addWhere('custom_group_id.is_multiple', '=', '0');
```
Comment (seamuslee001) [inline Civi/Api4/Service/Spec/SpecGatherer.php:126]: @colemanw do we need to do something about the multi record custom fields as well?

**Comment by guyiac** [comment]
Tested the second patch and worked fine.

1. Created search with Most Important Issue custom field.
2. disabled Most Important Issue custom field.
3. Field was gone from first search and didn't show up in a new one.
4. Created multiple value custom field set.
5. Created two new custom fields in set.
6. Verified both fields were available for search views.
7. Disabled one of the fields.
8. Verified that the field was no longer available for search views.

**Comment by demeritcowboy** [comment]
For reference there's a longstanding discussion about whether they should or shouldn't show: https://github.com/civicrm/civicrm-core/pull/14240#issuecomment-496791671 and also https://lab.civicrm.org/dev/core/-/issues/19. I suppose this PR matches where it is most places.

**Comment by colemanw** [comment]
Ok I went back and re-read that issue, and then double-checked the APIv3 behavior, and this PR matches it (disabled custom fields are not shown in APIv3 `getFields`). Based on that, I think this PR is ok to merge, but we could consider further refinements in the future.

**Pull request merged**